### PR TITLE
Schema spec for azure migrate service

### PR DESCRIPTION
**Pull request by vivekkothakota**
Schema spec for azure migrate service and test for the schema spec.

### CODE REVIEW

File: schemas/2018-02-02/Microsoft.Migrate.json [hunk 1]
```diff
@@ -0,0 +1,607 @@
+{
+  "id": "https://schema.management.azure.com/schemas/2018-02-02/Microsoft.Migrate.json#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Microsoft.Migrate",
+  "description": "Microsoft Migrate Resource Types",
+  "resourceDefinitions": {
+    "projects": {
```
Comment (TianoMS) [inline schemas/2018-02-02/Microsoft.Migrate.json]: >projects [](start = 5, length = 8)

please add the new resource types to schemas/2014-04-01-preview/deploymentTemplate.json and schemas/2015-01-01/deploymentTemplate.json.

Comment (vivekkothakota) [reply schemas/2018-02-02/Microsoft.Migrate.json]: Done. Thanks.

File: schemas/2015-01-01/deploymentTemplate.json [hunk 2]
```diff
@@ -610,6 +610,10 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_SOA" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_SRV" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_TXT" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2016-03-01/Microsoft.Insights.json#/resourceDefinitions/alertrules" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >{ "$ref": "https://schema.management.azure.com/schemas/2016-03-01/Microsoft.Insights.json#/resourceDefinitions/alertrules" }, [](start = 18, length = 125)

do you need this?

Comment (vivekkothakota) [reply schemas/2015-01-01/deploymentTemplate.json]: i did not add this line. It was there from earlier.
all i did was, add a comma at the end of the line.

Comment (Tiano2017) [reply schemas/2015-01-01/deploymentTemplate.json]: you added this line :). If you search it in the file, it now appears twice.

Comment (vivekkothakota) [reply schemas/2015-01-01/deploymentTemplate.json]: You are right. Sorry for the confusion. Thanks

**Comment by vivekkothakota** [comment]
Hi Tian, Anything else pending from my side?

**Pull request merged**